src/sql/workbench/contrib/welcome/page/browser/welcomePage.ts

```diff
@@ -0,0 +1,1118 @@
+			},
+			{
+				name: 'SQL Server Profiler',
+				link: 'https://docs.microsoft.com/sql/azure-data-studio/sql-server-profiler-extension?view=sql-server-2017',
+				icon: '../../../workbench/contrib/welcome/defaultExtensionIcon.svg'
+			},
+			{
+				name: 'SQL Server Import',
+				link: 'https://docs.microsoft.com/sql/azure-data-studio/sql-server-import-extension?view=sql-server-2017',
+				icon: '../../../workbench/contrib/welcome/defaultExtensionIcon.svg'
+			},
+			{
+				name: 'SQL Server Dacpac',
+				link: 'https://docs.microsoft.com/sql/azure-data-studio/sql-server-dacpac-extension?view=sql-server-2017',
+				icon: '../../../workbench/contrib/welcome/defaultExtensionIcon.svg'
+			}
+		],
+		isExtensionPack: true
+	},
+];
+
+const extensions: ExtensionSuggestion[] = [
+	{ name: localize('welcomePage.powershell', "Powershell"), id: 'microsoft.powershell', description: `Write and execute PowerShell scripts using Azure Data Studio's rich query editor`, icon: 'https://raw.githubusercontent.com/PowerShell/vscode-powershell/master/images/PowerShell_icon.png', link: `command:azdata.extension.open?{"id":"microsoft.powershell"}` },
+	{ name: localize('welcomePage.dataVirtualization', "Data Virtualization"), id: 'microsoft.datavirtualization', description: `Virtualize data with SQL Server 2019 and create external tables using interactive wizards`, icon: '../../../workbench/contrib/welcome/defaultExtensionIcon.svg', link: `command:azdata.extension.open?{"id":"microsoft.datavirtualization"}` },
+	{ name: localize('welcomePage.PostgreSQL', "PostgreSQL"), id: 'microsoft.azuredatastudio-postgresql', description: `Connect, query, and manage Postgres databases with Azure Data Studio`, icon: 'https://raw.githubusercontent.com/Microsoft/azuredatastudio-postgresql/master/images/extension-icon.png', link: `command:azdata.extension.open?{"id":"microsoft.azuredatastudio-postgresql"}` },
+];
+
+
+interface Strings {
+	installEvent: string;
+	installedEvent: string;
+	detailsEvent: string;
+
+	alreadyInstalled: string;
+	reloadAfterInstall: string;
+	installing: string;
+	extensionNotFound: string;
+}
+
+const extensionPackStrings: Strings = {
+	installEvent: 'installExtension',
+	installedEvent: 'installedExtension',
+	detailsEvent: 'detailsExtension',
+
+	alreadyInstalled: localize('welcomePage.extensionPackAlreadyInstalled', "Support for {0} is already installed."),
+	reloadAfterInstall: localize('welcomePage.willReloadAfterInstallingExtensionPack', "The window will reload after installing additional support for {0}."),
+	installing: localize('welcomePage.installingExtensionPack', "Installing additional support for {0}..."),
+	extensionNotFound: localize('welcomePage.extensionPackNotFound', "Support for {0} with id {1} could not be found."),
+};
+
+const tabKey = 9;
+const spaceKey = 32;
+const enterKey = 13;
+const escapeKey = 27;
+const upArrowKey = 38;
+const downArrowKey = 40;
+const leftArrowKey = 37;
+const rightArrowKey = 39;
+
+class KeyCodes {
```

Comment:
My above comment means this shouldn't be necessary - but in the future if you ever need to do something like this where you just need to have a bunch of values grouped together it's good practice and easier to just use a namespace. e.g.

```
namespace KeyCodes {
    export const spaceKey = 32;
    export const enterKey = 13;
    ...
}

let myVal = KeyCodes.spaceKey;
```

Having a class defined just for grouping values like this is unnecessary.

Reply:
resolved